wasm/text/decoder.go

```diff
@@ -18,17 +18,32 @@ const (
 	positionParam
 	positionResult
 	positionModule
-	positionModuleType
-	positionModuleImport
-	positionModuleImportFunc
-	positionModuleFunc
-	positionModuleMemory
-	positionModuleExport
-	positionModuleExportFunc
-	positionModuleExportMemory
-	positionModuleStart
+	positionImport
```

Comment:
a couple more refactors left (not now) and I'll be able to substitute sectionID for the top-level parsing position, and take these out.